### ISSUE TITLE
MS Teams O365 connectors will be deprecated

### ISSUE THREAD

**Issue by brainbug95** [automatiko-io/automatiko-approval-task#19]
Microsoft in all their foresight and wisdom decided to rather sudden [deprecate O365 connectors](https://aka.ms/O365ConnectorDeprecation) on which your amazing feature depends: [Teams](https://github.com/automatiko-io/automatiko-approval-task?tab=readme-ov-file#microsoft-teams)

Many projects are working to switch to alternate solutions (Workflows?). 

**Feature Request**

Do not use O365 connectors if possible.

**Comment by mswiderski** [comment]
Yes, already have this on my todo and should be worked on in coming weeks. Most likely based on workflows but will keep you posted on the progress

**Comment by brainbug95** [comment]
Hi @mswiderski 

It has been some time now since the issue was created. Did you have time to look into it? 

KR

**Comment by mswiderski** [comment]
true but the problem is there is no alternative to be used. All possible scenarios are way more complex and cannot be used as simple as current webhook based approach. Didn't MS move out of the deprecation?

**Comment by brainbug95** [comment]
They extended the deadline multiple times but now its gone:  https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/

_Update 4/14/2026: New information related to Office 365 Connectors Retirement in Teams – Final Deprecation_ 

I found that there is a Workflows template nowadays which kinda provides the same functionality.

"Send webhook alerts to a channel - Generate a webhook URL that lets anyone send external systems updates into a channel using rich adaptive cards or message cards."

https://support.microsoft.com/en-US/Workflows/send-messages-in-teams-using-incoming-webhooks

I got it working again. At least partially. The message gets send to my Teams Channel successfully but is missing the link to approval task button. I guess it is because the approval task is sending payload in an unsupported format.

Can you tell me where I can find/update the template used to send teams messages?

**Comment by mswiderski** [comment]
thanks for the info, if you manage to make this work updates to the template would be highly appreciated!

here is the template used for teams https://github.com/automatiko-io/automatiko-approval-task/blob/main/src/main/resources/templates/approval-teams.txt
